app/uk/gov/hmrc/gform/validation/ComponentValidator.scala

```diff
@@ -654,6 +673,169 @@ object ComponentValidator {
+      case WholeShape(_, _, _)                         => validationSuccess
+      case FractionalShape(_, _, _, _) if !isWhole     => validationSuccess
+      case _                                           => nonNumericSterlingFailure(fieldValue, value)
+    }
+  }
+
+  private def nonNumericSterlingFailure(
+    fieldValue: FormComponent,
+    value: String
+  )(implicit
+    messages: Messages,
+    sse: SmartStringEvaluator
+  ): ValidatedType[Unit] =
+    (fieldValue.errorShortName, fieldValue.errorShortNameStart, fieldValue.errorExample) match {
+      case (None, None, maybeErrorExample) =>
+        val errorExample = maybeErrorExample.getOrElse(SmartString.blank)
+        validationFailure(
+          fieldValue,
+          genericSterlingErrorPattern,
+          Some(List(SmartString.blank.trasform(_ => "an amount", _ => "swm").value(), errorExample.value()))
+        )
+
+      case (Some(errorShortName), None, maybeErrorExample) =>
+        val errorExample = maybeErrorExample.getOrElse(SmartString.blank)
+        validationFailure(
+          fieldValue,
+          genericSterlingErrorPattern,
+          Some(List(errorShortName.value(), errorExample.value()))
+        )
+      case (_, Some(errorShortNameStart), maybeErrorExample) =>
+        val errorExample = maybeErrorExample.getOrElse(SmartString.blank)
+        validationFailure(
+          fieldValue,
+          genericSterlingErrorPatternStart,
+          Some(List(errorShortNameStart.value(), errorExample.value()))
+        )
+    }
+
+  private def maxDigitSterlingFailure(
+    fieldValue: FormComponent,
+    value: String,
+    maxWhole: Int
+  )(implicit
+    messages: Messages,
+    sse: SmartStringEvaluator
+  ): ValidatedType[Unit] =
+    (fieldValue.errorShortNameStart, fieldValue.errorExample) match {
+      case (None, maybeErrorExample) =>
+        val errorExample = maybeErrorExample.getOrElse(SmartString.blank)
+        validationFailure(
+          fieldValue,
+          genericSterlingErrorMaxdigitPattern,
+          Some(
+            List(
+              SmartString.blank.trasform(_ => "Amount", _ => "swm").value(),
+              errorExample.value(),
+              maxWhole.toString
+            )
+          )
+        )
+      case (Some(errorShortNameStart), maybeErrorExample) =>
+        val errorExample = maybeErrorExample.getOrElse(SmartString.blank)
+        validationFailure(
+          fieldValue,
+          genericSterlingErrorMaxdigitPattern,
+          Some(
+            List(
+              errorShortNameStart.value(),
+              errorExample.value(),
+              maxWhole.toString
+            )
+          )
+        )
+    }
+
+  private def wholeSterlingFailure(
+    fieldValue: FormComponent,
+    value: String
+  )(implicit
+    messages: Messages,
+    sse: SmartStringEvaluator
+  ): ValidatedType[Unit] =
+    (fieldValue.errorShortNameStart, fieldValue.errorExample) match {
```

Comment:
All matching on `fieldValue.errorExample` are just rebinding the value, rather than doing any matching, can we drop that?